src/main/groovy/com/brightsparklabs/gradle/docs/DocsPlugin.groovy

```diff
@@ -467,14 +473,20 @@ public class DocsPlugin implements Plugin<Project> {
                      * be overridden in Asciidoc documents. See:
                      * - https://docs.asciidoctor.org/asciidoc/latest/attributes/assignment-precedence/
                      */
-                    attributes \
-                            'chapter-label@'      : '',
-                            'icons@'              : 'font',
-                            'imagesdir@'          : project.file(config.buildImagesDir),
-                            'numbered@'           : '',
-                            'source-highlighter@' : 'coderay',
-                            'toc@'                : config.tocPosition,
-                            'title-logo-image@'   : config.titleLogoImage
+
+                    Map<String,Object> pluginAttributes = [
+                        "chapter-label@"        : '',
+                        "icons@"                : 'coderay',
+                        "numbered@"             : '',
+                        "source-highlighter@"   : 'coderay',
+                        "toc@"                  : config.tocPosition,
+                        "title-logo-image@"     : config.titleLogoImage,
+                        "imagesdir@"            : project.file(config.buildImagesDir)
+                    ]
+                    pluginAttributes.putAll(config.attributes)
+                    // Allows for the removal of any attributes for which the user defines a value of null
+                    pluginAttributes.values().removeAll(Collections.singleton(null))
```

Comment:
I am surprised this removes the values from the map. Unless this a specific groovy thing, then it should not be doing so.
If you `println(pluginAttributes)` after this line, does that correctly show the values have been removed?

Reply:
It does function as expected, but I do concede that it does not make a lot of sense when you look at it. I have updated the implementation to something more modern and intuitive.